src/dowel/csv_output.py

```diff
@@ -41,13 +42,25 @@ def record(self, data, prefix=''):
                     extrasaction='ignore')
                 self._writer.writeheader()
 
-            if to_csv.keys() != self._fieldnames:
-                self._warn('Inconsistent TabularInput keys detected. '
-                           'CsvOutput keys: {}. '
-                           'TabularInput keys: {}. '
-                           'Did you change key sets after your first '
-                           'logger.log(TabularInput)?'.format(
-                               set(self._fieldnames), set(to_csv.keys())))
+            # add new keys to _fieldnames set
+            old_header_size = len(self._fieldnames)
+            for key in to_csv.keys():
+                self._fieldnames.add(key)
+
+            # if new keys are added, rewrite the csv file
+            if len(self._fieldnames) > old_header_size:
+                self._writer = csv.DictWriter(
```

Comment:
It's probably best to refactor out the writer initialization logic, so that if the parameters get changed somewhere they're consistently changed.

Reply:
Agree. Fixed in new commit.